src/types/query.d.ts

```diff
@@ -14,6 +14,9 @@ export module query {
   export function Object(fields: ExprArg): Expr;
   export function Lambda(f: Lambda): Expr;
   export function Lambda(var_name: ExprArg, expr: ExprArg): Expr;
+  export function Call(ref: ExprArg, ...args: ExprArg[]): Expr;
+  export function Query(lambda: Lambda): Expr;
+  export function Query(lambda: ExprArg): Expr;
```

Comment:
I don't think you need this overload considering the signature of `Map`, `Foreach`, and `Filter`. Since it's JS, I think the intent was to enforce the usage of native lambdas.

Reply:
this is a typescript thing, I think the compiler needs this overloaded version to type check the callers.

Reply:
If so, don't we need the same for the other functions that accept lambdas? Also, can't we use the `ExprArg|Lambda` syntax here?

Reply:
I'm asking because this function has a different signature than its similar siblings.

Reply:
well, removing this, ts users won't be able to to things like `Query(Lambda("x", Var("x")))` only `Query(x => x)`

Reply:
not that this is a problem :D

Reply:
I prefer the latter, looks cleaner. That's what I meant by enforcing native lambdas. Regardless of your choice, let's make sure every function that accepts lambda works the same.

Reply:
Ok, I removed that overloaded version